--- ../../dg_contrib/media_gallery/media_gallery.module	2016-02-26 21:15:49.908559711 -0600
+++ media_gallery.module	2016-02-26 21:18:15.524565375 -0600
@@ -372,6 +372,19 @@
   return $node;
 }
 
+ /**
+ * Implements hook_node_view()
+ * Ref: https://drupal.org/comment/5113982#comment-5113982
+ */
+function media_gallery_node_view($node, $view_mode, $langcode) {
+  /* Restore the media_gallery_media array which has been destroyed in
+   * media_gallery_view()
+   */
+  if (isset($node->media_gallery_media_original)) {
+    $node->media_gallery_media = $node->media_gallery_media_original;
+  }
+}
+
 /**
  * Implements hook_field_extra_fields().
  */
