diff --git a/media_browser_plus.module b/media_browser_plus.module
index da302dd..692d0b7 100644
--- a/media_browser_plus.module
+++ b/media_browser_plus.module
@@ -355,12 +355,13 @@ function media_browser_plus_get_media_root_folder($autocreate = FALSE) {
   $root_folder = FALSE;
   $vocabulary = taxonomy_vocabulary_machine_name_load('media_folders');
   if (!$vocabulary) {
+    $t = get_t();
     $vocabulary = (object) array(
       'name' => 'Media Folders',
-      'description' => st('Use media folders to organize your media'),
+      'description' => $t('Use media folders to organize your media'),
       'machine_name' => 'media_folders',
       'hierarchy' => 1,
-      'help' => st('Enter a concise name for the media folder'),
+      'help' => $t('Enter a concise name for the media folder'),
     );
     taxonomy_vocabulary_save($vocabulary);
   }
