diff --git a/addressfield_tokens.components.inc b/addressfield_tokens.components.inc
index db38907..62d6237 100644
--- a/addressfield_tokens.components.inc
+++ b/addressfield_tokens.components.inc
@@ -102,20 +102,15 @@ function _webform_render_addressfield($component, $value = NULL, $filter = TRUE)
   $available = !empty($component['extra']['available_countries']) ? $component['extra']['available_countries'] : NULL;
     
   // Get the current address
-  if (!empty($value[0])) {
-    if (is_string($value[0])) {
-      $address = unserialize($value[0]);
+  $address = _addressfield_tokens_expand_value($value);
+  if (empty($address)) {
+    if (!empty($component['value'])) {
+      $address = $component['value'];
     }
     else {
-      $address = $value[0];
+      $address = _webform_addressfield($component['cid']);
     }
   }
-  elseif (!empty($component['value'])) {
-    $address = $component['value'];
-  }
-  else {
-    $address = _webform_addressfield($component['cid']);
-  }
   if (empty($address)) {
     $address = addressfield_default_values($available);
   }
@@ -205,13 +200,7 @@ function _webform_addressfield_country_validate(&$element, &$form_state) {
  *   (such as wrapping the text) or as HTML (ensuring consistent output).
  */
 function _webform_display_addressfield($component, $value, $format = 'html') {
-  $address = NULL;
-  if (isset($value[0])) {
-    $address = $value[0];
-    if (is_string($address)) {
-      $address = unserialize($address);
-    }
-  }
+  $address = _addressfield_tokens_expand_value($value);
   return array(
     '#title'          => $component['name'],
     '#weight'         => $component['weight'],
@@ -220,7 +209,7 @@ function _webform_display_addressfield($component, $value, $format = 'html') {
     '#post_render'    => array('webform_element_wrapper'),
     '#component'      => $component,
     '#format'         => $format,
-    '#address'        => $address,
+    '#address'        => $address ? $address : NULL,
   );
 }
 
@@ -314,8 +303,9 @@ function _webform_analysis_addressfield($component, $sids = array(), $single = F
  *   Textual output formatted for human reading.
  */
 function _webform_table_addressfield($component, $value) {
-  if (!empty($value[0])) {
-    return theme('addressfield_formatter', array( 'address' => $value[0] ));
+  $address = _addressfield_tokens_expand_value($value);
+  if ($address) {
+    return theme('addressfield_formatter', array('address' => $address));
   }
   return '';
 }
@@ -374,17 +364,37 @@ function _webform_csv_headers_addressfield($component, $export_options) {
  *   every row of data.
  */
 function _webform_csv_data_addressfield($component, $export_options, $value) {
+  $address = _addressfield_tokens_expand_value($value);
   if (!empty($component['extra']['csv_separate']) && $component['extra']['csv_separate'] == 1) {
+    // Each address component should be in a separate column.
     $return = array();
     foreach (addressfield_tokens_property_names() as $key => $name) {
-      $return[] = (isset($value[0][$key])) ? $value[0][$key] : '';
+      $return[] = (isset($address[$key])) ? $address[$key] : '';
     }
     return $return;
   }
   else {
-    if (!empty($value[0])) {
-      return theme('addressfield_formatter__linear', array( 'address' => $value[0] ));
+    // The entire address should be displayed in the one column.
+    if ($address) {
+      return theme('addressfield_formatter__linear', array('address' => $address));
     }
     return '';
   }
 }
+
+/**
+ * Expand a raw address submission as loaded from the database to an array.
+ *
+ * @param string $value
+ *   An array of information containing the submission result, directly
+ *   correlating to the {webform_submitted_data} database schema.
+ *
+ * @return array|false
+ *   An associative array of address fields, or FALSE on failure.
+ */
+function _addressfield_tokens_expand_value($value) {
+  if (isset($value[0]) && is_string($value[0])) {
+    return unserialize($value[0]);
+  }
+  return FALSE;
+}
\ No newline at end of file
diff --git a/addressfield_tokens.module b/addressfield_tokens.module
index b804dce..b82df21 100644
--- a/addressfield_tokens.module
+++ b/addressfield_tokens.module
@@ -465,35 +465,6 @@ function _addressfield_tokens_webform_components($nid) {
 }
 
 /**
- * Implements hook_webform_submission_load().
- */
-function addressfield_tokens_webform_submission_load(&$submissions) {
-  $submissions_reset = reset($submissions);
-  $nid = $submissions_reset->nid;
-
-  $components = _addressfield_tokens_webform_components($nid);
-
-  foreach ($submissions as $sid => $submission) {
-    foreach ($components as $cid => $component) {
-      if (!empty($submission->data[$cid])) {
-        $parents = array($cid);
-        if (!empty($submission->data[$cid]['value'])) {
-          $parents[] = 'value';
-        }
-        $addresses = array();
-        foreach (drupal_array_get_nested_value($submission->data, $parents) as $delta => $data) {
-          $data = empty($data) ? array() : unserialize($data);
-          $addresses[$delta] = $data;
-        }
-        if (count($addresses)) {
-          drupal_array_set_nested_value($submission->data, $parents, $addresses);
-        }
-      }
-    }
-  }
-}
-
-/**
  * Implements hook_webform_validator_alter().
  */
 function addressfield_tokens_webform_validator_alter(&$validators) {
